src/Service.GraphQLBuilder/Sql/SchemaConverter.cs

```diff
@@ -31,184 +31,387 @@ public static class SchemaConverter
+                // object definitions of directional linking entities between (source, target) and (target, source).
+                // 2. The entity is not a linking entity and there is atleast one roles allowed to access the field.
+                if (rolesAllowedForFields.TryGetValue(key: columnName, out IEnumerable<string>? roles) || configEntity.IsLinkingEntity)
                 {
                     // Roles will not be null here if TryGetValue evaluates to true, so here we check if there are any roles to process.
-                    // Since Stored-procedures only support 1 CRUD action, it's possible that stored-procedures might return some values
-                    // during mutation operation (i.e, containing one of create/update/delete permission).
-                    // Hence, this check is bypassed for stored-procedures.
-                    if (roles.Count() > 0 || databaseObject.SourceType is EntitySourceType.StoredProcedure)
+                    // This check is bypassed for lnking entities for the same reason explained above.
+                    if (configEntity.IsLinkingEntity || roles is not null && roles.Count() > 0)
                     {
-                        if (GraphQLUtils.CreateAuthorizationDirectiveIfNecessary(
-                                roles,
-                                out DirectiveNode? authZDirective))
-                        {
-                            directives.Add(authZDirective!);
-                        }
-
-                        string exposedColumnName = columnName;
-                        if (configEntity.Mappings is not null && configEntity.Mappings.TryGetValue(key: columnName, out string? columnAlias))
-                        {
-                            exposedColumnName = columnAlias;
-                        }
-
-                        NamedTypeNode fieldType = new(GetGraphQLTypeFromSystemType(column.SystemType));
-                        FieldDefinitionNode field = new(
-                            location: null,
-                            new(exposedColumnName),
-                            description: null,
-                            new List<InputValueDefinitionNode>(),
-                            column.IsNullable ? fieldType : new NonNullTypeNode(fieldType),
-                            directives);
-
+                        FieldDefinitionNode field = GenerateFieldForColumn(configEntity, columnName, column, directives, roles);
                         fields.Add(columnName, field);
                     }
                 }
             }
 
-            if (configEntity.Relationships is not null)
+            // A linking entity is not exposed in the runtime config file but is used by DAB to support nested mutations on entities with M:N relationship.
+            // Hence we don't need to process relationships for the linking entity itself.
```

Comment:
You point out what is not done for a linking entity. It would be helpful if this comment described what the next code block is actually doing.